packages/utils/src/cmds/schema/diff/index.js

```diff
@@ -81,18 +89,50 @@ function sortArrayProps(obj) {
     })
 }
 
+function transformArrayToObject(array, hashKey) {
+    return array.reduce((acc, curr, index) => {
+        let hash = curr[hashKey]
+        if (!hash) {
+            reporter.warn(
+                'hash-candidate value not found, falling back to index for ',
+                curr
+            )
+            hash = index
+        }
+        acc[curr[hashKey]] = curr
```

Comment:
I think this needs to be `acc[hash]` in order to follow the logic in the if statement (if the hash candidate isn't found) - otherwise `curr[hashKey]` will be undefined here and so the different array values will overwrite eachother

Reply:
cc: @Birkbjo

Reply:
@Birkbjo pinging this one again

Reply:
Sorry for the super late reply. This has drowned in my inbox. But you are right, will fix.